go/parquet/internal/utils/math.go

```diff
@@ -0,0 +1,45 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+func Min(a, b int64) int64 {
```

Comment:
there aren't standard library functions that do this arleady?

Reply:
Unfortunately Go doesn't have generics, (if all goes well, they will get introduced in Go 1.18!) so these are basically just convenience functions as the stdlib Min/Max functions only take float64 and this allows me to not have to cast to float64 everywhere when i want to use Min/Max. It actually ends up being common practice in go code and you can even see this in the Go arrow package for the same reason. When generics are finally introduced into Go we can get rid of this.